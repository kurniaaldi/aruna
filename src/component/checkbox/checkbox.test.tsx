import { render, screen } from "@testing-library/react";
import Checkbox from "./index";
import renderer from "react-test-renderer";

test("dialog", () => {
  let values = {
    name: "input1",
    check: false,
  };

  const handleChange = (event: any) => {};

  render(
    <Checkbox
      checked={values.check}
      name={values.name}
      onChange={handleChange}
    />
  );

  const checkboxElement = screen.getByTestId("input1");
  expect((checkboxElement as HTMLInputElement).checked).toBe(false);
  expect((checkboxElement as HTMLInputElement).checked).toBe(false);
});

test("input-snapshoot", () => {
  let values = {
    name: "input1",
    check: false,
  };

  const handleChange = (event: any) => {};

  const tree = renderer
    .create(
      <Checkbox
        checked={values.check}
        name={values.name}
        onChange={handleChange}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
