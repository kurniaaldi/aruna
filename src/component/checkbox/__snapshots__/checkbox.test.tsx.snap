// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input-snapshoot 1`] = `
<input
  checked={false}
  data-testid="input1"
  name="input1"
  onChange={[Function]}
  style={
    Object {
      "height": 20,
      "width": 20,
    }
  }
  type="checkbox"
/>
`;
